lib/manager/jenkins/extract.ts

```diff
@@ -1,12 +1,69 @@
+import is from '@sindresorhus/is';
+import yaml from 'js-yaml';
 import * as datasourceJenkins from '../../datasource/jenkins-plugins';
 import { logger } from '../../logger';
 import { SkipReason } from '../../types';
 import { isSkipComment } from '../../util/ignore';
 import * as dockerVersioning from '../../versioning/docker';
 import type { PackageDependency, PackageFile } from '../types';
 
-export function extractPackageFile(content: string): PackageFile | null {
-  logger.trace('jenkins.extractPackageFile()');
+function getDependency(
+  name: string,
+  version: string,
+  url: string,
+  groupId: string
+): PackageDependency {
+  const dep: PackageDependency = {
+    datasource: datasourceJenkins.id,
+    versioning: dockerVersioning.id,
+    depName: name,
+  };
+
+  if (version) {
+    dep.currentValue = version.toString();
+  } else {
+    dep.skipReason = SkipReason.NoVersion;
+  }
+
+  if (version === 'latest' || version === 'experimental' || groupId) {
+    dep.skipReason = SkipReason.UnsupportedVersion;
+  }
+
+  if (url) {
+    dep.skipReason = SkipReason.InternalPackage;
+  }
+
+  logger.debug({ dep }, 'Jenkins plugin dependency');
+  return dep;
+}
+
+function extractYaml(content: string): PackageDependency[] {
+  const deps: PackageDependency[] = [];
+
+  try {
+    const doc = yaml.safeLoad(content, { json: true }) as any;
```

Comment:
Please create types and add to `./types.ts`

Reply:
Can you point me to an example of what you would like it to look like ? other usages of `yaml.safeLoad` seemed to use `as any`.

Reply:
https://github.com/renovatebot/renovate/blob/c8f105ab968a9e4ed2c4efea2dcda0d782ab8eae/lib/manager/gitlabci/extract.ts#L103-L105

😏

Reply:
Done! Thanks :)